hooks/useTransactionSender.tsx

```diff
@@ -50,54 +358,67 @@ export const useTransactionSender = () => {
 
       try {
         const signedTxs = await wallet.signAllTransactions(timedTxs.flat());
-        const signatures = [];
 
         // Reconstruct signed sequence
-        const signedSequence: T[][] = [];
+        const signedSequence: TransactionInfo<T>[][] = [];
         let i = 0;
         sequence.forEach((set) => {
-          const signedSet: T[] = [];
-          set.forEach(() => {
-            signedSet.push(signedTxs[i]);
-            i += 1;
+          const signedSet: TransactionInfo<T>[] = [];
+          set.forEach((el) => {
+            const transactionSignature = signedTxs[i].signatures[0];
+            const signature =
+              transactionSignature instanceof Uint8Array
+                ? Buffer.from(transactionSignature)
+                : transactionSignature.signature;
+
+            if (signature) {
+              signedSet.push({
+                signature: base58.encode(signature),
+                transaction: {
+                  ...el,
+                  tx: signedTxs[i],
+                },
+              });
+
+              i += 1;
+            }
           });
           signedSequence.push(signedSet);
         });
 
-        // eslint-disable-next-line no-restricted-syntax
-        for (const set of signedSequence) {
-          signatures.push(
-            // eslint-disable-next-line no-await-in-loop
-            ...(await Promise.all(
-              set.map((tx) =>
-                connection
-                  .sendRawTransaction(tx.serialize(), {
-                    skipPreflight: true,
-                  })
-                  .then((txSignature) =>
-                    connection.confirmTransaction(txSignature).then(() => txSignature),
-                  ),
-              ),
-            )),
-          );
-        }
-
-        notifications.show({
-          title: 'Transactions sent!',
-          message: (
-            <Stack>
-              {signatures.map((signature) => (
-                <NotificationLink key={signature} signature={signature} />
-              ))}
-            </Stack>
+        // Send signed transactions
+        const signatures = await Promise.all(
+          signedSequence.flatMap((set) =>
+            set.map((el) =>
+              connection.sendRawTransaction(el.transaction.tx.serialize(), {
+                skipPreflight: true,
+              }),
+            ),
           ),
-          autoClose: 5000,
+        );
+
+        const transactionInfos = signedSequence.flat();
+
+        // render an initial set of transactions in a single notification
+        const id = notifications.show({
```

Comment:
Just curious, what is this id value?

Reply:
It's the unique ID for a given notification instance, generated when you call the `show` function. This allows us to later update a specific notification. From my testing, it usually looks something like `mantine-<random_alphanumeric_chars>`

https://mantine.dev/x/notifications/#notification-props

Reply:
@jshiohaha ahhh okay so it's from the mantime lib it's not from the chain itself right? Maybe down the road we could tie things to the blockchain txn ID instead so when other's use it the library is separate from the UI components. Although, maybe this part of the code is really specific to our use case and the abstraction doesn't make sense here, either way, just something for future thought.

Reply:
It's funny you say this because I actually had a bunch of different mappings to help manage renderings and updates — signature to notification ID, notification ID to signature, etc. It ended up feeling a bit overkill because it was a lot of updating these mappings. 

Although, it was probably just bad design/implementation on my side, so I ended up here for managing different notifications 😅

Reply:
Yeah well there are so many different ways to do it, it can get confusing I think. I think overall though this will improve the current UI a lot and we'll want to have something very similar in the new UI/SDK. Great work!